test/StreamJsonRpc.Tests/MarshalableProxyTests.cs

```diff
@@ -1040,6 +1141,57 @@ public Task AcceptGenericProxyContainerAsync(ProxyContainer<IGenericMarshalable<
         public Task AcceptNonMarshalableAsync(INonMarshalable nonMarshalable) => Task.CompletedTask;
 
         public Task AcceptNonMarshalableDerivedFromMarshalablesAsync(INonMarshalableDerivedFromMarshalable nonMarshalable) => Task.CompletedTask;
+
+        public async Task CallScopedMarshalableAsync(IMarshalableWithCallScopedLifetime marshalable)
+        {
+            await marshalable.DoSomethingAsync();
+
+            this.ContinuationResult = Task.Run(async delegate
+            {
+                await this.AllowContinuation.WaitAsync();
+                await marshalable.DoSomethingAsync();
+            });
+        }
+
+        public Task<IMarshalableWithCallScopedLifetime> ReturnCallScopedObjectAsync()
+        {
+            // Returning a call-scoped object as a return type is illegal.
+            // This method is used in a test that verifies the failure mode for this case.
+            return Task.FromResult<IMarshalableWithCallScopedLifetime>(new MarshalableAndSerializable());
+        }
+
+        public async IAsyncEnumerable<int> CallScopedMarshalableReturnsAsyncEnumerable(IMarshalableWithCallScopedLifetime marshalable)
+        {
+            // Yield before using the marshalable since we want to test that the call-scoped argument
+            // is available for the whole async enumerable state machine.
+            await Task.Yield();
+
+            await marshalable.DoSomethingAsync();
```

Comment:
Does "await Task.Yield" behave same as "yield return <somevalue>"? 
I was going to suggest doing "yield return .." twice with calling DoSomethingAsync in between for completeness sake.

Reply:
No, `await Task.Yield();` does not actually produce a value, where `yield return` does. 
The test was effective without yielding two values, and it keeps the unit test method a bit simpler to produce just one.  TBH I don't even know if I needed the yield first, because async enumerables aren't executed at all until they are actually enumerated, which in the case of RPC doesn't happen by default until the receiver starts the foreach.